source/agora/flash/Node.d

```diff
@@ -853,10 +853,11 @@ public abstract class FlashNode : ControlFlashAPI
         // our own secret (we are the payee)
         if (auto secret = payment_hash in this.secrets)
         {
-            assert(!error); // Payee should never fail to receive
-            channel.learnSecrets([*secret], [], this.last_block_height);
+            if (!error)
```

Comment:
I can't see how, is there an example you can give? AFAICS, `onPaymentComplete` is never called with an error for incoming payments anyways.

Reply:
The signing process can fail. I added a notification for failed incoming payments, but now that I think about it maybe it should be a separate callback.

I'll rework it a bit.